server/routes/product.js

```diff
@@ -0,0 +1,158 @@
+const express = require("express");
+const router = express.Router();
+const { body, validationResult } = require("express-validator");
+const mongoose = require("mongoose");
+
+const User = require("../models/User");
+const Product = require("../models/Product");
+
+// @route     POST api/product
+// @desc      Register a product
+// @access    Public
+router.post(
+    "/",
+    [
+        body("title", "Please add title").not().isEmpty(),
+        body("price", "Please enter price").not().isEmpty(),
+        body("user_id", "Please enter user_id").not().isEmpty()
+    ],
+    async (req, res) => {
+        const errors = validationResult(req);
+        if (!errors.isEmpty()) {
+            return res.status(400).json({ errors: errors.array() });
+        }
+
+        const { title, description, price, user_id, photos } = req.body;
+
+        try {
+            let user = await User.findById(user_id);
```

Comment:
since this never changes I would suggest turning this variable into a constant `const`